runtimevar/example_test.go

```diff
@@ -28,38 +28,44 @@ import (
 	"google.golang.org/grpc/status"
 )
 
-func Example_constantvarJsonVariable() {
-	// DBConfig is the sample config struct we're going to parse our JSON into.
-	type DBConfig struct {
-		Host     string
-		Port     int
-		Username string
+func Example_jsonDecoder() {
+	// This example is used in https://gocloud.dev/howto/runtimevar/runtimevar/#awsps-ctor
+
+	// Variables set up elsewhere:
+	ctx := context.Background()
+
+	// Config is the sample config struct we're going to parse our JSON into.
+	type Config struct {
+		Host string
+		Port int
 	}
 
-	// Here's our sample JSON config.
-	const jsonConfig = `{"Host": "gocloud.dev", "Port": 8080, "Username": "testuser"}`
+	// A sample JSON config that will decode into Config.
+	const jsonConfig = `{"Host": "gocloud.dev", "Port": 8080}`
 
-	// We need a Decoder that decodes raw bytes into our config.
-	decoder := runtimevar.NewDecoder(DBConfig{}, runtimevar.JSONDecode)
+	// Construct a Decoder that decodes raw bytes into our config.
+	decoder := runtimevar.NewDecoder(Config{}, runtimevar.JSONDecode)
 
-	// Next, a construct a *Variable using a constructor from one of the
-	// runtimevar subpackages. This example uses constantvar.
+	// Next, a construct a *Variable using a constructor or URL opener.
+	// This example uses constantvar.
+	// If you're using a URL opener, you can't decode JSON into a struct, but
+	// you can use the query parameter "decoder=jsonmap" to decode into a map.
 	v := constantvar.NewBytes([]byte(jsonConfig), decoder)
 	defer v.Close()
+	// snapshot.Value will be of type Config.
 
-	// Call Latest to retrieve the value.
-	snapshot, err := v.Latest(context.Background())
+	// Ignore unused variables in example:
```

Comment:
I don't see where you ignore them. I am assuming you refer to  using _. But here you actually use each variable in the code below.

Reply:
Similar to above. The comment is being used to trim the example shown on `gocloud.dev`, while allowing the rest of it to be shown in godoc and to actual run.